site/components/TransactionsModal.js

```diff
@@ -0,0 +1,146 @@
+import useTranslation from 'next-translate/useTranslation'
+
+import { EtherscanLink } from './EtherscanLink'
+import JustifiedBetweenRow from './JustifiedBetweenRow'
+import Modal from './Modal'
+import SvgContainer from './svg/SvgContainer'
+import { useNumberFormat } from '../hooks/useNumberFormat'
+
+const TransactionModalRow = function ({ text, value, tipLink = '' }) {
+  return (
+    <JustifiedBetweenRow
+      keyComponent={
+        <p
+          className={`text-gray-350 text-sm ${
+            tipLink ? 'flex justify-bewtween cursor-pointer' : ''
+          }`}
+          onClick={() => tipLink && window.open(tipLink, '_blank')}
+        >
+          {text}
+          {tipLink && <SvgContainer className="ml-1" name="questionmark" />}
+        </p>
+      }
+      valueComponent={<p className="text-sm font-semibold">{value}</p>}
+    />
+  )
+}
+
+// eslint-disable-next-line complexity
+const TransactionsModal = function ({ transaction, modalIsOpen, closeModal }) {
+  const { t } = useTranslation('common')
+  const formatNumber = useNumberFormat()
+
+  const isConfirmed = transaction.transactionStatus === 'confirmed'
+  const isError =
+    transaction.transactionStatus === 'canceled' ||
+    transaction.transactionStatus === 'error'
+
+  return (
+    <Modal
+      className="relative flex flex-col w-full h-screen bg-white border-0 shadow-lg outline-none max-w-screen-sm focus:outline-none md:rounded-lg md:h-auto"
+      modalIsOpen={modalIsOpen}
+      onRequestClose={closeModal}
+    >
+      <div className="p-6">
+        {/* Header: operation name and closing "x" */}
+        <div className="border-b">
+          <button className="float-right" onClick={closeModal}>
+            <SvgContainer name="close" />
+          </button>
+          <p className="mb-2 font-bold text-left">
+            {t(`${transaction.operation}`)}
+          </p>
+        </div>
+        <div className="mt-4">
+          {/* Values sent and received */}
+          {transaction.sent && (
+            <div className="flex items-center justify-between pb-2 text-lg font-bold border-b border-gray-300">
+              <div className="w-2/5 text-left">
+                {`${formatNumber(transaction.sent)} ${transaction.sentSymbol}`}
+              </div>
+              <div className="w-1/5 text-2xl">→</div>
+              <div className="w-2/5 text-right">
+                {transaction.received.map((token, i) => (
+                  <div key={i}>{`${formatNumber(token.value)} ${
+                    token.symbol
+                  }`}</div>
+                ))}
+              </div>
+            </div>
+          )}
+
+          {/* General operation information: total txs, fees, status */}
+          <div className="py-4">
+            <TransactionModalRow
+              text={t('total-transactions')}
+              value={transaction.suffixes && transaction.suffixes.length}
```

Comment:
if `transaction.suffixes` is undefined (and I guess it could be because you're checking that it is defined), wouldn't this render `undefined?`

Reply:
Yeah. `transactions.suffixes` should always be defined.

Reply:
Changed in b73e76b.